source/Nevermore.IntegrationTests/Advanced/UseExistingTransactionFixture.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using System.Data;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.Data.SqlClient;
+using Nevermore.Advanced;
+using Nevermore.IntegrationTests.SetUp;
+using NUnit.Framework;
+
+namespace Nevermore.IntegrationTests.Advanced;
+
+#pragma warning disable NV0008 // Nevermore transaction is never disposed - false positives
+
+public class UseExistingTransactionFixture : FixtureWithRelationalStore
+{
+    CancellationToken cancellationToken;
+
+    public override void SetUp()
+    {
+        base.SetUp();
+        cancellationToken = new CancellationTokenSource(TimeSpan.FromMinutes(1)).Token;
+    }
+
+    [Test]
+    public async Task OpeningReadTransaction_FromExistingConnectionAndTransaction_Throws()
+    {
+        await using var sqlConnection = new SqlConnection(ConnectionString);
+        await sqlConnection.OpenAsync(cancellationToken);
+        await using var sqlTransaction = sqlConnection.BeginTransaction();
+
+        var readTransaction = (ReadTransaction) Store.CreateReadTransactionFromExistingConnectionAndTransaction(sqlConnection, sqlTransaction);
+
+        using (new AssertionScope())
+        {
+            const string expectedMessage = "An existing connection and transaction were provided, they should have been opened externally";
+
+            await ((Func<Task>)(async () => await readTransaction.OpenAsync(cancellationToken)))
+                .Should()
+                .ThrowExactlyAsync<InvalidOperationException>()
+                .WithMessage(expectedMessage);
+
+            ((Action)(() => readTransaction.Open()))
```

Comment:
Trivia: Rather than casting the lambda to Action you can do `new Action(() => readTransaction.Open()).Should()...` which has fewer brackets